command/agent/local.go

```diff
@@ -379,6 +379,12 @@ func (l *localState) setSyncState() error {
 		}
 
 		// If our definition is different, we need to update it
+		if existing.EnableTagDrift {
+			l.logger.Printf("[DEBUG] Tag drift enabled.")
+			existing.Tags = service.Tags
```

Comment:
@ryanuber don't we require that the local data changes to match the server if drift is enabled so that we don't sync back to the old tags if the service definition changes for some other reason? Also, this is the `localState` which just has maps vs. memdb. Want to make sure I understand your concern before we pull the trigger on this one.